tests/rules/no-unsafe-takeuntil.ts

```diff
@@ -0,0 +1,228 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { stripIndent } from "common-tags";
+import rule = require("../../source/rules/no-unsafe-takeuntil");
+import { ruleTester } from "../utils";
+
+ruleTester({ types: true }).run("no-unsafe-takeuntil", rule, {
+  valid: [
+    {
+      code: stripIndent`
+        import { of } from "rxjs";
+        import { switchMap, takeUntil } from "rxjs/operators";
+
+        const a = of("a");
+        const b = of("b");
+        const c = of("d");
+
+        const d = a.pipe(switchMap(_ => b), takeUntil(c)).subscribe();
+      `
+    },
+    {
+      code: stripIndent`
+        import { combineLatest, of } from "rxjs";
+        import { takeUntil } from "rxjs/operators";
+
+        const a = of("a");
+        const b = of("b");
+        const c = of("c");
+        const d = of("d");
+
+        const e = a.pipe(s => combineLatest(s, b, c), takeUntil(d)).subscribe();
+      `
+    },
+    {
+      code: stripIndent`
+        import { of } from "rxjs";
+        import { switchMap, takeUntil } from "rxjs/operators";
+
+        const a = of("a");
+        const b = of("b");
+        const c = of("d");
+
+        const d = a.pipe(switchMap(_ => b), pipe(takeUntil(d), switchMap(_ => b))).subscribe();
```

Comment:
I think we might want to see this is an error?
rxjs-tslint-rules doesn't see this an error, so for now I left it as is.

Reply:
I don't mind. This is an edge case that's unlikely to occur. It's in the tests, IIRC, to show that the rule's implementation does not care about this usage.